packages/namekit-react/src/components/DomainExpirationWarning.tsx

```diff
@@ -1,55 +1,90 @@
 import {
-  Address,
   AvailableNameTimelessPriceUSD,
-  DomainCard,
-  formattedPrice,
   formatTimestampAsDistanceToNow,
+  SecondaryRegistrationStatus,
   PrimaryRegistrationStatus,
   domainExpirationTimestamp,
-  SecondaryRegistrationStatus,
-  formatTimestamp,
   domainReleaseTimestamp,
   UserOwnershipOfDomain,
   getUserOwnership,
+  formatTimestamp,
+  formattedPrice,
+  Registration,
+  ENSName,
+  Address,
 } from "@namehash/ens-utils";
 import { ClockIcon, ClockUrgency } from "./icons/ClockIcon";
-import { Normalization } from "@namehash/nameguard";
 import React, { useEffect, useState } from "react";
 import { Tooltip } from "./Tooltip";
 import cc from "classcat";
 
 interface DomainExpirationWarningProps {
-  onlyIcon?: boolean;
+  domainName: ENSName | null; // when null we show the skeleton
+  domainRegistration: Registration | null;
   /* 
-    Below address is used to display different text
-    messages based on the ownership of the domain
+    If domain is in Grace Period, the address that used to own 
+    the domain is considered as the former owner of the domain
+
+    More reference documentation can be found in:
+    https://support.ens.domains/en/articles/8046877-name-lifecycle
   */
-  viewerAddress: Address | null;
+  formerDomainOwnerAddress: Address | null;
+  /* 
+    If domain is Registered or Expired, there is one
+    address that is considered the owner of the domain
+
+    More reference documentation can be found in:
+    https://support.ens.domains/en/articles/8046877-name-lifecycle
+  */
+  currentDomainOwnerAddress: Address | null;
+  /* 
+    Below address is the one used in domain's ownership
+    determination, where depending on the match between
+    the "addressToCompareOwnership", "currentDomainOwnerAddress"
+    and the "formerDomainOwnerAddress", different Ui elements
+    are displayed alongside with text information about
+    the specific domain's ownership status.
+  */
+  addressToCompareOwnership: Address | null;
```

Comment:
I think first option is more generic since I can imagine that is not in our hands the comparison being done with an dApp current user but I get your idea behind the name proposed and am happy to make it more specific to the main use case we are providing support

Reply:
@FrancoAguzzi Yes this logic and the returned data structure is all relative to the current user, not some abstract "addressToCompareOwnership".